models/post_repo.go

```diff
@@ -0,0 +1,75 @@
+package models
+
+import "fmt"
+
+// PostRepository handles the CRUD for post.
+type PostRepository interface {
+	Create(*Post) error
+	Get(*Query) (*Post, error)
+	GetAll(*Query) (*[]Post, error)
+	Update(*Post) error
+}
+
+// DBPostRepository ...
+type DBPostRepository struct{}
+
+// NewDBPostRepository ...
+func NewDBPostRepository() PostRepository {
+	return DBPostRepository{}
+}
+
+// Create handles the creation of post.
+func (repo DBPostRepository) Create(post *Post) error {
+	tx := DB.MustBegin()
+	_, err := tx.NamedExec("INSERT INTO post (body, caption, user_id, ctime, utime) VALUES (:body, :caption, :user_id, :ctime, :utime)", post)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// Get handles the fetching of a single post record.
+func (repo DBPostRepository) Get(query *Query) (*Post, error) {
+	post := Post{}
+
+	queryString := "SELECT * FROM post"
+
+	if query.ByColumn != nil {
+		queryString = queryString + fmt.Sprintf(" WHERE %v = %v", query.ByColumn.Column, query.ByColumn.Value)
+	}
+
```

Comment:
So instead of removing the **ByColumn** I removed the **ByID** Clause so we can query other columns not just the **ID**. @kenkoii @rbo13